### ISSUE TITLE
lost connection: Internal Exception

### ISSUE THREAD

**Issue by RRixxles** [ViaVersion/ViaVersion#3639]
### `/viaversion dump` Output

https://dump.viaversion.com/d3ddb85de07d328ad8f2ad7fdc3229e7a6b5503594b9f3463c86d45afadcfc21

### Console Error

```
[Put the controlc.com link or text here.](https://controlc.com/6a9815ee)
```

### Bug Description

I'm trying to join my 1.20.4 server on 1.20.1 (also tried 1.20.2) and i'm getting this error. I've tried updating to a new version of viaversion & viabackwards and it just doesn't work at all if I do. So I'm really unsure what the problem is.

### Steps to Reproduce

1. Install viaversion & viabackwards from the spigotmc link
2. make your server paper 1.20.4 and try to join on 1.20.1 (havent tested other versions)

### Expected Behavior

allow the player to join

### Additional Server Info

Using paper with these listed plugins:

![image](https://github.com/ViaVersion/ViaVersion/assets/155375340/668e788d-6b77-427e-9b78-8b0da08aa8e9)
![image](https://github.com/ViaVersion/ViaVersion/assets/155375340/3666f322-7fa1-44af-87db-eaee4359488e)
![image](https://github.com/ViaVersion/ViaVersion/assets/155375340/ddd0d769-5f5b-48b3-8ed9-08a643847941)

### Checklist

- [X] Via plugins are only running on **EITHER** the backend servers (e.g. Paper) **OR** the proxy (e.g. BungeeCord), **not on both**.
- [X] I have included a ViaVersion dump.
- [X] If applicable, I have included a paste (**not a screenshot**) of the error.
- [X] I have tried the latest build(s) from https://ci.viaversion.com/ and the issue still persists.

**Comment by Barvalg** [comment]
![Platform: git--Paper--336%20%28MC%3A%201.20.4%29](https://img.shields.io/badge/Platform-git--Paper--336%20%28MC%3A%201.20.4%29-3498db?style=flat-square) </br>![ViaVersion (4.9.2): 30 commits behind master](https://img.shields.io/badge/ViaVersion%20(4.9.2)-30%20commit(s)%20behind%20master-yellow?style=flat-square) </br>![ViaBackwards(4.9.1): 17 commits behind master](https://img.shields.io/badge/ViaBackwards%20(4.9.1)-17%20commit(s)%20behind%20master-yellow?style=flat-square) </br><br><h3>Please update **all** Via* plugins from [https://ci.viaversion.com/](https://ci.viaversion.com/). In case the issue still persists send the new dump and the issue will be reopened</h3>

**Comment by Kichura** [comment]
Side note: Minehut controls the viaversion jar which makes it nearly impossible to resolve your problem properly.

**Comment by RRixxles** [comment]
> Side note: Minehut controls the viaversion jar which makes it nearly impossible to resolve your problem properly.

Wdym?

**Comment by Kichura** [comment]
1. You can't override their own viaversion jar provided by them with yours,
2. Minehut has been known for using old viaversion copies which breaks viabackwards.

**Comment by RRixxles** [comment]
> Side note: Minehut controls the viaversion jar which makes it nearly impossible to resolve your problem properly.

It worked properly before, but I added a data pack to my server in the server.properties button, then it starting doing that error.

I loaded a backup but the error still persists

**Comment by RRixxles** [comment]
> 1. You can't override their own viaversion jar provided by them with yours,
> 2. Minehut has been known for using old viaversion copies which breaks viabackwards.

If I get the same version of viabackwards as minehut's viaversion, would that change anyrthing?

**Comment by Kichura** [comment]
No since the one provided in their marketplace is even older *(4.2.0)* and because the resource pack fix only works properly in viaversion 4.9.3-snapshot and viabackwards 4.9.2-snapshot which in this case is not possible as you would have to ask minehut to update the viaversion jar for you before hand,

But if you absolutely require server resource packs to work then i suggest using Aternos or another server host provider for this even if it sounds unfortunate.

**Comment by RRixxles** [comment]
> No since the one provided in their marketplace is even older _(4.2.0)_ and because the resource pack fix only works properly in viaversion 4.9.3-snapshot and viabackwards 4.9.2-snapshot which in this case is not possible as you would have to ask minehut to update the viaversion jar for you before hand,
> 
> But if you absolutely require server resource packs to work then i suggest using Aternos or another server host provider for this even if it sounds unfortunate.

I'm still unsure why it worked a few hours ago and now it doesn't. Is that a problem with minehut?

**Comment by Kichura** [comment]
Has been for a while which we can't fix.
